pkg/serverutils/knative_event_source_utils.go

```diff
@@ -0,0 +1,75 @@
+type CustomResourceDefinitionSpec struct {
+	Group   string                        `json:"group" protobuf:"bytes,1,opt,name=group"`
+	Version string                        `json:"version,omitempty" protobuf:"bytes,2,opt,name=version"`
+	Names   CustomResourceDefinitionNames `json:"names" protobuf:"bytes,3,opt,name=names"`
+}
+
+// CustomResourceDefinitionNames indicates the names to serve this CustomResourceDefinition
+type CustomResourceDefinitionNames struct {
+	Plural     string   `json:"plural" protobuf:"bytes,1,opt,name=plural"`
+	Singular   string   `json:"singular,omitempty" protobuf:"bytes,2,opt,name=singular"`
+	ShortNames []string `json:"shortNames,omitempty" protobuf:"bytes,3,opt,name=shortNames"`
+	Kind       string   `json:"kind" protobuf:"bytes,4,opt,name=kind"`
+	ListKind   string   `json:"listKind,omitempty" protobuf:"bytes,5,opt,name=listKind"`
+	Categories []string `json:"categories,omitempty" protobuf:"bytes,6,rep,name=categories"`
+}
+
+//TypeMeta describes type of the object and its API schema version.
+type TypeMeta struct {
+	Kind       string `json:"kind,omitempty" protobuf:"bytes,1,opt,name=kind"`
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,2,opt,name=apiVersion"`
+}
+
+// ObjectMeta is metadata that all persisted resources must have, which includes all objects users must create
+type ObjectMeta struct {
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+}
+
+// CustomResourceDefinition represents a resource that should be exposed on the API server.
+type CustomResourceDefinition struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Spec       CustomResourceDefinitionSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// ListMeta describes metadata that resources must have, including lists and various status objects
+type ListMeta struct {
+	SelfLink        string `json:"selfLink,omitempty" protobuf:"bytes,1,opt,name=selfLink"`
+	ResourceVersion string `json:"resourceVersion,omitempty" protobuf:"bytes,2,opt,name=resourceVersion"`
+}
+
+// CustomResourceDefinitionList is a list of CustomResourceDefinition objects.
+type CustomResourceDefinitionList struct {
+	Kind       string `json:"kind,omitempty" protobuf:"bytes,1,opt,name=kind"`
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,2,opt,name=apiVersion"`
+	ListMeta   `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// items list individual CustomResourceDefinition objects
+	Items []CustomResourceDefinition `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// KnativeEventSourceFilter shall filter partial metadata from knative event sources CRDs before propagating
+func KnativeEventSourceFilter(w http.ResponseWriter, r io.Reader) {
+	var typedRes CustomResourceDefinitionList
+
+	if err := json.NewDecoder(r).Decode(&typedRes); err != nil {
+		plog.Errorf("Knative Event Source CRD response deserialisation Failed: %s", err)
```

Comment:
not sure where the `plog` is coming from ?

Reply:
```suggestion
		plog.Errorf("Knative Event Source CRD response deserialization failed: %s", err)
```

Reply:
`plog` is already defined in `serverutils` package, utils.go